release-channels-and-staged-deployments/README.md

```diff
@@ -0,0 +1,379 @@
+    - Faster feedback loop when working on MC components, which can be especially important in KaaS Area.
+    - Less control over upgrades, bugs can get deployed to many (all) clusters before catching them.
+2. Manually triggered deployments
+    - We use this approach mostly for WCs.
+    - We use this approach for updating vintage WCs to newer releases.
+    - This way we and the customers have more control over the cluster upgrades.
+    - The control over upgrades can be especially important for some customers' production clusters where we also have freeze periods where we must not do updates.
+    - It can be a problem when security and critical bug fixes have to be rolled out as soon as possible.
+
+#### 1.2.1. Automated continuous deployments
+
+Giant Swarm today in vintage:
+- We continuously push some apps to app collections and those are being then deployed automatically to all MCs. (MC)
+
+Giant Swarm today in CAPI:
+- We continuously push some apps to app collections and those are being then deployed automatically to all MCs. (MC)
+- We continuously deploy some of cluster-x apps to some management clusters with renovate automatically merging PRs. (WC, MC)
+- We continuously deploy some of default-apps-x apps to some management clusters with renovate automatically merging PRs. (WC, MC)
+
+#### 1.2.2. Manual deployments
+
+Today in vintage:
+- We and our customers manually update vintage WCs to newer releases.
+
+Giant Swarm today in CAPI:
+- We and our customers manually update clusters to newer cluster-x versions.
+- We and our customers manually update clusters to newer default-apps-x versions.
+
+
+## 2. Next step: release channels and staged deployments
+
+This sections describes the new way for defining releases and deploying components. Most of the existing concepts and processes are kept, and then they are expanded and placed into new concepts.
+
+The first new concept that we define here is release channels. Release channels differ in stability levels and release cycle frequency. All components are available in all release channels.
+
+The second new concept that we define here are staged deployments. All components are delivered to all release channels in stages. We also define how we promote component versions (and other cluster changes) from one release channel to another.
+
+From the above, we can say that the staged deployments consist of gradually promoting changes from one release channel to another with stability and release cycle frequency in mind.
+
+Now let's go deeper.
+
+
+### 2.1. Release channels
+
+For release channels we define 3 stability levels, each of which has one or more release channels:
+- Testing
+- Staging
+- Production
+
+For production stability level we define 3 release channels, which have different release cycles, but all are considered to be production:
+- Production Edge - fast rolling releases
+- Production Current - normal point releases
+- Production LTS - slow point releases
```

Comment:
I dont like the naming here since it can be very confusing with customer env naming. In the end, we will not force customers to set a real `dev` env cluster as `Testing` release channel. So example a customer can set `Testing` channel in their `Staging` cluster if they consider to that channel as appropriate to the staging cluster. I would advocate for something along the lines of `alpha`, `beta` and `stable` instead of.